airflow/jobs/backfill_job_runner.py

```diff
@@ -245,7 +245,16 @@ def _update_counters(self, ti_status: _DagRunTaskStatus, session: Session) -> No
             session.execute(
                 update(TI)
                 .where(filter_for_tis)
-                .values(state=TaskInstanceState.SCHEDULED)
+                .values(
+                    state=TaskInstanceState.SCHEDULED,
+                    try_number=case(
+                        (
+                            or_(TI.state.is_(None), TI.state != TaskInstanceState.UP_FOR_RESCHEDULE),
```

Comment:
Do we need the or if you reverse it? (Positive check on up_for_reschedule)

Reply:
you do need it because sql.

Reply:
to be clear i thought it should work with positive check on up for...

i.e.`where not state = 'up_for_reschedule'` but it did not work. i don't recall if maybe sqlalchemy was just being too "helpful" or what.

snowflake has `equal_null` function that handles this case.  alas we don't run on snowflake 🤷